Poker.BE/Poker.BE.Domain/Security/UserManager.cs

```diff
@@ -4,72 +4,89 @@
+			UsersDictionary.Add(userName, UserToAdd);
+			return UserToAdd;
+		}
 
-        protected bool RemoveUser (string userName){
-            if (CheckExistingUser (userName)){
-                UsersDictionary.Remove(userName);
-                return true;
-            }
-            return false;
-        }
+		protected bool RemoveUser(string userName)
+		{
+			if (CheckExistingUser(userName))
+			{
+				UsersDictionary.Remove(userName);
+				return true;
+			}
+			return false;
+		}
 
-        protected bool CheckExistingUser (string userName){
-            if (userName != null)
-                return (UsersDictionary.ContainsKey(userName));
-            return false;
-        }
+		protected bool CheckExistingUser(string userName)
+		{
+			if (userName != null)
+				return (UsersDictionary.ContainsKey(userName));
+			return false;
+		}
 
-        
-        protected bool CheckPasswordValidity (string password){
-            if (password.Length >= 6 ) return true;
-            return false;
-        }
 
-        public User LogIn (string userName, string password){
-            if (CheckExistingUser(userName)){ // We check that the user is existing in our DB
-               User UserToCheck;
-               if (UsersDictionary.TryGetValue(userName, out UserToCheck) ){ // We take the User Object from our DB
-                   string GoodPassword = UserToCheck.Password;
-                   bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
-                   if (arePasswordMatching) {
-                       UserToCheck.Connect();
-                       return UserToCheck;
-                    }
-               }
-            }
-            return null;
-        }
+		protected bool CheckPasswordValidity(string password)
+		{
+			if (password.Length >= 6) return true;
+			return false;
+		}
 
-        public bool LogOut (User userToLogout){
-            userToLogout.Disconnect();
-            return true;
-        }
+		public User LogIn(string userName, string password)
+		{
+			if (!CheckExistingUser(userName)) // We check that the user is existing in our DB
+				throw new UserNotFoundException();
+			User UserToCheck;
+			if (UsersDictionary.TryGetValue(userName, out UserToCheck))
+			{ // We take the User Object from our DB
+				string GoodPassword = UserToCheck.Password;
+				bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
+				if (!arePasswordMatching)
+				{
+					throw new IncorrectPasswordException();
+				}
+				UserToCheck.Connect();
+				return UserToCheck;
+			}
+			return null;
+		}
 
+		public bool LogOut(User userToLogout)
+		{
+			if (!CheckExistingUser(userToLogout.UserName)) // We check that the user is existing in our DB
+				throw new UserNotFoundException();
+			userToLogout.Disconnect();
+			return true;
+		}
 
-        #endregion
 
-    }
+		#endregion
+
+	}
```

Comment:
i will need more explanation of this code over the phone. please add comments and `///<summery>` over methods. use `#regions` (also regions inside of regions) and everything that will make your code easier to understand (especially formatting - enough said about that... 😃 ). at this moment, i found your code very hard to understand.